core/Scheduler/Scheduler.php

```diff
@@ -147,6 +152,33 @@ public function run()
                     $readFromOption = true;
                     $message = $this->executeTask($task);
 
+                    // Task has thrown an exception and should be scheduled for a retry
```

Comment:
just to be safe does it make sense to call `$this->scheduleRetry = false;` before `$message = $this->executeTask($task);` just in case it's not reset? Or maybe we reset it directly after the `if ($this->scheduleRetry) {`? Just some thought. It might work now with no problem, but it would make it less likely to break in the future depending on what gets refactored